clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -5800,6 +5827,18 @@ void CGDebugInfo::setDwoId(uint64_t Signature) {
 }
 
 void CGDebugInfo::finalize() {
+  for (auto const *VD : StaticDataMemberDefinitionsToEmit) {
+    assert(VD->isStaticDataMember());
+
+    if (DeclCache.contains(VD))
+      continue;
+
+    if (!VD->hasInit())
```

Comment:
it's non-obvious why this skip exists, can you add a comment here?

Reply:
I'll update the `EmitGlobalVariable` docs to clarify that we need an initialiser, otherwise behaviour is undefined